extensions/amp-access/0.1/amp-access-server-jwt.js

```diff
@@ -128,28 +144,116 @@ export class AccessServerJwtAdapter {
         this.serverState_,
         this.clientAdapter_.getAuthorizationUrl());
     if (!this.isProxyOrigin_ || !this.serverState_) {
-      dev().fine(TAG, 'Proceed via client protocol');
-      return this.clientAdapter_.authorize();
+      return this.authorizeOnClient_();
     }
+    return this.authorizeOnServer_();
+  }
 
-    dev().fine(TAG, 'Proceed via server protocol');
+  /** @override */
+  pingback() {
+    return this.clientAdapter_.pingback();
+  }
 
-    const varsPromise = this.context_.collectUrlVars(
+  /**
+   * @return {!Promise<{encoded:string, jwt:!JSONObject}>}
+   * @private
+   */
+  fetchJwt_() {
+    const urlPromise = this.context_.buildUrl(
         this.clientAdapter_.getAuthorizationUrl(),
         /* useAuthData */ false);
-    return varsPromise.then(vars => {
-      const requestVars = {};
-      for (const k in vars) {
-        if (vars[k] != null) {
-          requestVars[k] = String(vars[k]);
-        }
+    let jwtPromise = urlPromise.then(url => {
+      dev().fine(TAG, 'Authorization URL: ', url);
+      return this.timer_.timeoutPromise(
+          AUTHORIZATION_TIMEOUT,
+          this.xhr_.fetchText(url, {
+            credentials: 'include',
+            requireAmpResponseSourceOrigin: true,
+          }));
+    }).then(encoded => {
+      const jwt = this.jwtHelper_.decode(encoded);
+      user().assert(jwt['amp_authdata'],
+          '"amp_authdata" must be present in JWT');
+      return {encoded, jwt};
+    });
+    if (this.shouldBeValidated_()) {
+      // Validate JWT in the development mode.
+      if (this.jwtHelper_.isVerificationSupported()) {
+        jwtPromise = jwtPromise.then(resp => {
+          return this.jwtHelper_.decodeAndVerify(resp.encoded, this.keyUrl_)
+              .then(() => resp);
+        });
+      } else {
+        user().warn(TAG, 'Cannot verify signature on this browser since' +
+            ' it doesn\'t support WebCrypto APIs');
       }
+      jwtPromise = jwtPromise.then(resp => {
+        this.validateJwt_(resp.jwt);
+        return resp;
+      });
+    }
+    return jwtPromise.catch(reason => {
+      throw user().createError('JWT fetch or validation failed: ', reason);
+    });
+  }
+
+  /**
+   * @return {boolean}
+   * @private
+   */
+  shouldBeValidated_() {
+    return getMode().development;
+  }
+
+  /**
+   * @param {!JSONObject} jwt
+   * @private
+   */
+  validateJwt_(jwt) {
+    const now = Date.now();
+
+    // exp: expiration time.
+    const exp = jwt['exp'];
+    user().assert(exp, '"exp" field must be specified');
+    user().assert(parseFloat(exp) * 1000 > now,
+        'token has expired: %s', exp);
+
+    // aud: audience.
+    const aud = jwt['aud'];
```

Comment:
True. I found the reference to this. Implemented an array as well. Thanks for the catch!